apollo-runtime/src/main/java/com/apollographql/apollo/subscription/OperationClientMessage.java

```diff
@@ -62,26 +62,48 @@ public Init(@NotNull Map<String, Object> connectionParams) {
     private static final String JSON_KEY_QUERY = "query";
     private static final String JSON_KEY_VARIABLES = "variables";
     private static final String JSON_KEY_OPERATION_NAME = "operationName";
+    private static final String JSON_KEY_EXTENSIONS = "extensions";
+    private static final String JSON_KEY_EXTENSIONS_PERSISTED_QUERY = "persistedQuery";
+    private static final String JSON_KEY_EXTENSIONS_PERSISTED_QUERY_VERSION = "version";
+    private static final String JSON_KEY_EXTENSIONS_PERSISTED_QUERY_HASH = "sha256Hash";
```

Comment:
Should this just be queryHashMethod? It's not necessarily sha256

Reply:
that part is really interesting, GraphQL spec doesn't have any info regarding this. BTW I noticed the same issue with regular queries, we sending `sha256Hash ` but as you said it can be different (md5). As far as I remember Apollo server expects `sha256Hash`. 

What about your use case, what GraphQL server are you using? Are you using sha256 as hash? 

I guess the only option right now is to make it configurable.

Reply:
@sav007 . This part is already merged, but would it be better to just name it `hash` as it could be anything. 

We use md5